### PR TITLE
Fix reading bool value with CRLF in CSV format

### LINKED ISSUE #44401: Parsing of boolean values with CRLF from CSV inputs fails
**Describe what's wrong**

Parsing of boolean values with CRLF from CSV inputs fails.

**Does it reproduce on recent release?**

Yes

**How to reproduce**

```
CREATE TABLE string_test (col1 String) Engine=Memory
```

```
CREATE TABLE boolean_test (col1 Boolean) Engine=Memory
```

```
❯ echo -ne "True\r\nTrue\r\n" | clickhouse client --query="INSERT INTO string_test FORMAT CSV"
```

```
❯ echo -ne "True\r\nTrue\r\n" | clickhouse client --query="INSERT INTO boolean_test FORMAT CSV"
Code: 467. DB::Exception: Cannot parse boolean value here: 'True
True', should be 'true' or 'false' controlled by setting bool_true_representation and bool_false_representation or one of True/False/T/F/Y/N/Yes/No/On/Off/Enable/Disable/Enabled/Disabled/1/0: (at row 1)
:
Row 1:
Column 0,   name: col1, type: Bool, ERROR: text "True<CARRIAGE RETURN><LINE FEED>True" is not like Bool

: While executing ParallelParsingBlockInputFormat: data for INSERT was parsed from stdin: (in query: INSERT INTO boolean_test FORMAT CSV). (CANNOT_PARSE_BOOL)
```

**Expected behavior**

Should be able to insert booleans with CRLF.

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix reading bool value with CRLF in CSV format.
Closes https://github.com/ClickHouse/ClickHouse/issues/44401

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by nsweeting** [comment]
The test doesn't seem to touch the broken behaviour right now:

```
❯ echo -ne "True\r\nFalse\r\n" | clickhouse local --input-format=CSV -q "select * from table";
True
False
```

vs

```
❯ echo -ne "True\r\nFalse\r\n" | clickhouse local --structure "a Boolean, b Boolean" --input-format=CSV -q "select * from table";
Code: 467. DB::Exception: Cannot parse boolean value here: 'True
Fals', should be 'true' or 'false' controlled by setting bool_true_representation and bool_false_representation or one of True/False/T/F/Y/N/Yes/No/On/Off/Enable/Disable/Enabled/Disabled/1/0: (at row 1)
:
Row 1:
Column 0,   name: a, type: Bool, ERROR: text "True<CARRIAGE RETURN><LINE FEED>Fals" is not like Bool

: While executing ParallelParsingBlockInputFormat: While executing File. (CANNOT_PARSE_BOOL)
```

**Comment by Avogar** [comment]
I forgot to add `--structure='x Bool'` to the command:
```
echo -ne "True\r\nFalse\r\n" | clickhouse local --structure='a Boolean' --input-format=CSV -q "select * from table"
Code: 467. DB::Exception: Cannot parse boolean value here: 'True
Fals', should be 'true' or 'false' controlled by setting bool_true_representation and bool_false_representation or one of True/False/T/F/Y/N/Yes/No/On/Off/Enable/Disable/Enabled/Disabled/1/0: (at row 1)
:
Row 1:
Column 0,   name: x, type: Bool, ERROR: text "True<CARRIAGE RETURN><LINE FEED>Fals" is not like Bool
```

I'll fix that.

Your second example contains incorrect CSV data. `"True\r\nFalse\r\n"` cannot be parsed with structure `"a Boolean, b Boolean"`

File: tests/queries/0_stateless/02513_csv_bool_allow_crlf.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+echo -ne "True\r\nFalse\r\n" | $CLICKHOUSE_LOCAL --stracture='x Bool' --input-format=CSV -q "select * from table";
```
Comment (nsweeting) [inline tests/queries/0_stateless/02513_csv_bool_allow_crlf.sh:7]:
```suggestion
echo -ne "True\r\nFalse\r\n" | $CLICKHOUSE_LOCAL --structure='x Bool' --input-format=CSV -q "select * from table";
```

**Review by nsweeting — CHANGES_REQUESTED** [review]

File: tests/queries/0_stateless/02513_csv_bool_allow_crlf.sh (hunk 1 above)
Comment (Avogar) [reply tests/queries/0_stateless/02513_csv_bool_allow_crlf.sh]: Oups :)

**Review by yakov-olkhovskiy — APPROVED** [review]

File: tests/queries/0_stateless/02513_csv_bool_allow_crlf.reference [hunk 2]
```diff
@@ -0,0 +1,2 @@
+True
+False
```
Comment (nsweeting) [inline tests/queries/0_stateless/02513_csv_bool_allow_crlf.reference:2]:
```suggestion
true
false
```

**Review by nsweeting — CHANGES_REQUESTED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**